src/test/regress/specs/isolation_hash_copy_vs_all.spec

```diff
@@ -0,0 +1,96 @@
+step "s1-insert" { INSERT INTO hash_copy VALUES(0, 'k'); }
+step "s1-update" { UPDATE hash_copy SET data = 'l' WHERE id = 0; }
+step "s1-delete" { DELETE FROM hash_copy WHERE id = 1; }
+step "s1-truncate" { TRUNCATE hash_copy; }
+step "s1-drop" { DROP TABLE hash_copy; }
+step "s1-ddl-create-index" { CREATE INDEX hash_copy_index ON hash_copy(id); }
+step "s1-ddl-drop-index" { DROP INDEX hash_copy_index; }
+step "s1-ddl-add-column" { ALTER TABLE hash_copy ADD new_column int DEFAULT 0; }
+step "s1-ddl-drop-column" { ALTER TABLE hash_copy DROP new_column; }
+step "s1-ddl-rename-column" { ALTER TABLE hash_copy RENAME data TO new_data; }
+step "s1-table-size" { SELECT citus_table_size('hash_copy'); SELECT citus_relation_size('hash_copy'); SELECT citus_total_relation_size('hash_copy'); }
+step "s1-master-modify-multiple-shards" { SELECT master_modify_multiple_shards('DELETE FROM hash_copy;'); }
+step "s1-create-non-distributed-table" { CREATE TABLE hash_copy(id integer, data text); COPY hash_copy FROM PROGRAM 'echo 0, a\\n1, b\\n2, c\\n3, d\\n4, e' WITH CSV; }
+step "s1-distribute-table" { SELECT create_distributed_table('hash_copy', 'id'); }
+step "s1-commit" { COMMIT; }
+
+# session 2
+session "s2"
+step "s2-copy" { COPY hash_copy FROM PROGRAM 'echo 5, f\\n6, g\\n7, h\\n8, i\\n9, j' WITH CSV; }
+step "s2-copy-additional-column" { COPY hash_copy FROM PROGRAM 'echo 5, f, 5\\n6, g, 6\\n7, h, 7\\n8, i, 8\\n9, j, 9' WITH CSV; }
+step "s2-router-select" { SELECT * FROM hash_copy WHERE id = 1; }
+step "s2-multi-shard-select" { SELECT * FROM hash_copy ORDER BY id, data; }
+step "s2-insert" { INSERT INTO hash_copy VALUES(0, 'k'); }
+step "s2-update" { UPDATE hash_copy SET data = 'l' WHERE id = 0; }
+step "s2-delete" { DELETE FROM hash_copy WHERE id = 1; }
+step "s2-truncate" { TRUNCATE hash_copy; }
+step "s2-drop" { DROP TABLE hash_copy; }
+step "s2-ddl-create-index" { CREATE INDEX hash_copy_index ON hash_copy(id); }
+step "s2-ddl-drop-index" { DROP INDEX hash_copy_index; }
+step "s2-ddl-create-index-concurrently" { CREATE INDEX CONCURRENTLY hash_copy_index ON hash_copy(id); }
+step "s2-ddl-add-column" { ALTER TABLE hash_copy ADD new_column int DEFAULT 0; }
+step "s2-ddl-drop-column" { ALTER TABLE hash_copy DROP new_column; }
+step "s2-ddl-rename-column" { ALTER TABLE hash_copy RENAME data TO new_data; }
+step "s2-table-size" { SELECT citus_table_size('hash_copy'); SELECT citus_relation_size('hash_copy'); SELECT citus_total_relation_size('hash_copy'); }
+step "s2-master-modify-multiple-shards" { SELECT master_modify_multiple_shards('DELETE FROM hash_copy;'); }
+step "s2-create-non-distributed-table" { CREATE TABLE hash_copy(id integer, data text); COPY hash_copy FROM PROGRAM 'echo 0, a\\n1, b\\n2, c\\n3, d\\n4, e' WITH CSV; }
+step "s2-distribute-table" { SELECT create_distributed_table('hash_copy', 'id'); }
+
+# permutations - COPY vs COPY
+permutation "s1-begin" "s1-copy" "s2-copy" "s1-commit" "s1-select-count"
```

Comment:
They do not block each other even while processing on same shard. I think it is OK to leave it this way.